src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/BehandlingService.kt

```diff
@@ -45,25 +48,29 @@ class BehandlingService(
     fun lagBehandlingRespons(behandlingId: Long): BehandlingResponsDto {
         val behandling = hentBehandling(behandlingId)
         val arbeidsfordelingPåBehandling = arbeidsfordelingService.hentArbeidsfordelingPåBehandling(behandlingId)
+        val personopplysningGrunnlag = personopplysningGrunnlagService.hentAktivPersonopplysningGrunnlag(behandlingId)
 
-        val personer =
-            personopplysningGrunnlagService.hentAktivPersonopplysningGrunnlag(behandlingId)?.personer?.toList()
-                ?: emptyList()
-
+        val personer = personopplysningGrunnlag?.personer?.toList() ?: emptyList()
         val landKodeOgLandNavn = personer.flatMap { it.statsborgerskap }.toSet()
             .associate { it.landkode to statsborgerskapService.hentLand(it.landkode) }
-
-        val personResponserDtoer = personer.map { lagPersonRespons(it, landKodeOgLandNavn) }
+        val personResponser = personer.map { lagPersonRespons(it, landKodeOgLandNavn) }
 
         val søknadsgrunnlag = søknadGrunnlagService.finnAktiv(behandlingId)?.tilSøknadDto()
         val personResultater =
             vilkårsvurderingService.finnAktivVilkårsvurdering(behandlingId)?.personResultater?.toList()
+
+        val andelerTilkjentYtelse = andelTilkjentYtelseRepository.finnAndelerTilkjentYtelseForBehandling(behandlingId)
+        val personerMedAndelerResponsDto =
+            personopplysningGrunnlag?.let { lagPersonerMedAndelTilkjentYtelseRespons(it.personer, andelerTilkjentYtelse) }
+                ?: emptyList()
```

Comment:
Ikke noe veldig big deal, men kunne vi sendt inn `personopplysningGrunnlag` og `andelerTilkjentYtelse` som parametere inn i `lagBehandlingRespons` og utledet DTO'en der inne slik vi gjør for de andre feltene?

Reply:
Ja, det kan gjøres. Men denne metoden blir ganske stor og allerede har 7 parametere som metode argumenter. Så tenker å sende dto-en slik at vi sender  så få parametere som mulig.